dags/python_scripts/cr3_extract_diagram_ocr_narrative.py

```diff
@@ -167,151 +190,185 @@ def update_crash_metadata(crash_id: int, metadata: dict) -> bool:
+for crash in response.json()["data"]["atd_txdot_crashes"]:
+    if args.v:
+        print("Preparing to operate on crash_id: " + str(crash["crash_id"]))
 
     # build url and download the CR3
-    if (args.v):
-        print('Pulling CR3 PDF from S3');
-    key = args.cr3_source[1] + '/' + str(crash['crash_id']) + '.pdf'
+    if args.v:
+        print("Pulling CR3 PDF from S3")
+    key = args.cr3_source[1] + "/" + str(crash["crash_id"]) + ".pdf"
     obj = []
     try:
         pdf = s3.get_object(Bucket=args.cr3_source[0], Key=key)
     except:
-        sys.stderr.write("Error: Failed to get PDF from the S3 object\n")
+        sys.stderr.write(f"Error: Failed to get PDF for crash {str(crash['crash_id'])}from the S3 object\n")
         continue
 
     # render the pdf into an array of raster images
-    if (args.v):
-        print('Rendering PDF into images');
+    if args.v:
+        print("Rendering PDF into images")
     pages = []
     try:
-        pages = convert_from_bytes(pdf['Body'].read(), 150)
-    except: 
-        sys.stderr.write("Error: PDF Read for crash_id (" + str(crash['crash_id']) + ") failed.\n")
+        pages = convert_from_bytes(pdf["Body"].read(), 150)
+    except:
+        sys.stderr.write(
+            "Error: PDF Read for crash_id (" + str(crash["crash_id"]) + ") failed.\n"
+        )
         continue
 
-
-    if (args.d):
-        if (args.v):
-            print('Excuting a check for a digitally created PDF');
+    if args.d:
+        if args.v:
+            print("Executing a check for a digitally created PDF")
         digital_end_to_end = True
-        # these pixels are expected to be black on digitally created PDFs
-        pixels = [(110,3520), (3080, 3046), (3050, 2264), (2580, 6056), (1252, 154), (2582, 4166), (1182, 1838)]
-        for pixel in pixels:
+        # assume new cr3 form unless proven otherwise
+        new_cr3_form = True
+        # these pixels are expected to be black on digitally created PDFs previous to 1/1/2023
+        pixels = [
+            (110, 3520),
+            (3080, 3046),
+            (3050, 2264),
+            (2580, 6056),
+            (1252, 154),
+            (2582, 4166),
+            (1182, 1838),
+        ]
+        new_pixels = [
+            (215, 2567),
+            (872, 2568),
+            (3338, 3060),
+            (1690, 2574),
+            (4834, 279)
+        ]
+        for pixel in new_pixels:
             rgb_pixel = pages[1].getpixel(pixel)
-            if not(rgb_pixel[0] == 0 and rgb_pixel[1] == 0 and rgb_pixel[2] == 0):
-                digital_end_to_end = False
-            if (args.v):
-                print('Pixel' + "(%04d,%04d)" % pixel + ': ' + str(rgb_pixel))
-        if (args.v):
-            print('PDF Digital End to End?: ' + str(digital_end_to_end));
-        if not(digital_end_to_end):
-            if (args.v):
+            if not (rgb_pixel[0] == 0 and rgb_pixel[1] == 0 and rgb_pixel[2] == 0):
+                new_cr3_form = False
```

Comment:
if these aren't black then we know its not the new cr3 form and we should check to see if its the old one